dulwich/repo.py

```diff
@@ -140,10 +140,12 @@ def _get_default_identity() -> Tuple[str, str]:
     import getpass
     import socket
 
-    username = getpass.getuser()
```

Comment:
I believe the USER environment variable should be present on Windows - see https://pureinfotech.com/list-environment-variables-windows-10/. Where are you finding that it's missing?

Reply:
getpass should always be available IIUC